wolfcrypt/src/port/nrf51.c

```diff
@@ -74,18 +74,12 @@ int nrf51_random_generate(byte* output, word32 size)
     }
 
     while (remaining > 0) {
-        err_code = nrf_drv_rng_bytes_available(&available);
-        if (err_code == NRF_SUCCESS) {
-            length = (remaining < available) ? remaining : available;
-            if (length > 0) {
-                err_code = nrf_drv_rng_rand(&output[pos], length);
-                remaining -= length;
-                pos += length;
-            }
-        }
-
-        if (err_code != NRF_SUCCESS) {
```

Comment:
If its because of this line, to me it looks like this block should be above below the `nrf_drv_rng_rand` call... At some point the err_code result needs to get checked.